lib/index.js

```diff
@@ -74,15 +121,40 @@ function compileBody(str, options){
         return filename;
       },
       read: function (filename) {
+        var str;
+        var plugins = getPlugin('loadRead');
+        for (var i = 0; i < plugins.length; i++) {
+          str = plugins[i](filename, source, options, dependencies);
+          if (str) {
+            debug_sources[filename] = str;
+            return str;
+          }
+        }
         dependencies.push(filename);
```

Comment:
We need to allow the plugin some way to **optionally** indicate a dependency on a file.